httpd/filters.go

```diff
@@ -0,0 +1,147 @@
+package httpd
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+
+	"github.com/influxdb/influxdb"
+)
+
+// basicAuthCredentials returns the username and password encoded in
+// a request. The credentials may be present as URL query params, or as
+// a Basic Authentication header.
+func basicAuthCredentials(r *http.Request) (string, string, error) {
```

Comment:
We should support both. Passing via parameters is a usability issue. Parameters are easier to do it through curl and they let you just toss it into your browser and let your JSON prettier make it look good.